fixed #29

stephanebastian opened the pull request:
This is a WIP. textfield-filled is 90% done. outlined is completely broken

@robinlahtinen : This is the first stab at fixing #29 . 
It's currently working on Firefox and Chrome. I believe it should be working with fairly old browsers and the default mode is 'populated' so that old browsers don't supporting fancy css would work.
Note that I changed classnames to be 'textfield-' instwerd of 'fieldtext-'  and such. 
Whenever you have time I would appreciate any feedback you may have.

robinlahtinen reviewed (changes requested):
Currently `dist` folder should only be updated on new releases. Reason being, that [the website](https://www.materialdesignlight.com/) uses `main` as a production branch and automatically updates from it.

robinlahtinen reviewed (commented):
I'll do a proper full review later, but from a quick glance (both code and running it) the current changes look good and the code is clean!

stephanebastian commented:
@robinlahtinen  I believe this PR fixes #29. Could you please do a full review whenever you have some spare time ?
As a side note, you will see that I've added the file _md.scss and updated the import section. When I got started on implementing this feature, I found that it was somewhat a pain to prefix theme related variables with 'light', or state related variables with state.$varName (same goes for typo and such). So I added the file md.scss. The goal is to import only this file (@use 'md' as *) so we reference variables as $md-sys-typescale-body-large-size.
It's not a big deal but I found it easier when doing a cut/paste of variables from the material design spec. Wondering if you´d feel the same. Cheers

stephanebastian commented:
Not sure why the labeler is failing. Might be related to this issue though: https://github.com/actions/labeler/issues/12
it seems to be a permission issue. 
Do you have an idea on how to fix it?

robinlahtinen commented on scss/_fields.scss:
```diff
@@ -5,406 +5,547 @@
+    
+      // label - by default it's styled in 'populated mode' so that browsers not supporting 'advanced' css are ok
+      & > .#{$classname}-label {
+        background-color: white;
```
Label background colour should be inherited from the background. Probably best way would be to include background in outlined text field and possibility to change it with general background classes, or perhaps try `background: inherit;` on the whole element.

robinlahtinen commented on site/src/index.njk:
```diff
@@ -84,39 +84,95 @@ keywords:
-              class='material-symbols-outlined material-symbols-outlined-2'>cancel</span></label>
+    <div class='row'>
+      <div class='col'>
+        <span>Filled with supporting text</span>
```
Document spans should use proper typography class. Check out the `_typography.scss` for classes. It doesn't currently include all variants, but I'll add them. Best class would probably be `typography-body-large`.

robinlahtinen commented:
> I found that it was somewhat a pain to prefix theme related variables with 'light' [...] It's not a big deal but I found it easier when doing a cut/paste of variables from the material design spec.

Good idea! Material Design Light supports dark theme, so that should also be taken in to concern.

> Could you please do a full review whenever you have some spare time ?

Code looks good and seems to be up to the specifications mostly. It is missing support for dark theme and some lesser problems I referenced directly on above code reviews.

When hovering mouse over the text field, the cursor doesn't change the type to `text` (shape of I-beam) mode on the inner sides.

stephanebastian commented:
@robinlahtinen I've made the changes you requested, splitted the file in two files, one for filled text fields and another one for outlined text fields. Could you please review one more time and merge if its ok? Thx
